website/static/js/registrationUtils.js

```diff
@@ -0,0 +1,802 @@
+RegistrationEditor.prototype.save = function() {
+    // TODO(lyndsysimon): Test this.
+    var self = this;
+    var request;
+
+    var payload = self.serialized();
+    if (typeof self.draft().pk === 'undefined') {
+        // Draft has not yet been assigned a PK; create a new DraftRegistration
+        request = $osf.postJSON(self.urls.create, payload);
+        request.done(function(data) { self.draft().pk = data.pk; }.bind(self));
+    } else {
+        // Update an existing DraftRegistration
+        var url = self.urls.update.replace('{draft_pk}', self.draft().pk);
+        request = $osf.putJSON(url, payload);
+    }
+
+    self.lastSaveRequest(request);
+
+    request.done(function(data, status, xhr) {
+        xhr.completedDate = new Date();
+        xhr.payload = payload;
+        // Explicitly set the observable to tell KO it changed.
+        self.lastSaveRequest(xhr);
+    });
+
+    return request;
+};
+
+RegistrationEditor.prototype.saveForLater = function () {
+    var self = this;
+
+    self.save();
+    window.location = self.urls.draftRegistrations;
+};
+
+/**
+ * @class RegistrationManager
+ * Model for listing DraftRegistrations
+ *
+ * @param {Object} node: optional data to instatiate model with
+ * @param {String} draftsSelector: DOM node to bind VM to
+ * @param {Object} urls:
+ * @param {String} urls.list:
+ * @param {String} urls.get:
+ * @param {String} urls.delete:
+ * @param {String} urls.edit:
+ * @param {String} urls.schemas:
+ **/
+var RegistrationManager = function(node, draftsSelector, urls) {
+    var self = this;
+
+    self.node = node;
+    self.draftsSelector = draftsSelector;
+
+    self.urls = urls;
+
+    self.schemas = ko.observableArray();
+    self.selectedSchema = ko.observable();
+
+    // TODO: convert existing registration UI to frontend impl.
+    // self.registrations = ko.observable([]);
+    self.drafts = ko.observableArray();
+    self.hasDrafts = ko.pureComputed(function() {
+        return self.drafts().length > 0;
+    });
+
+    self.loading = ko.observable(true);
+
+    self.preview = ko.observable(false);
+
+    // bound functions
+    self.getDraftRegistrations = $.getJSON.bind(null, self.urls.list);
+    self.getSchemas = $.getJSON.bind(null, self.urls.schemas);
+};
+RegistrationManager.prototype.init = function() {
+    var self = this;
+
+    $osf.applyBindings(self, self.draftsSelector);
+
+    var getSchemas = self.getSchemas();
+
+    getSchemas.then(function(response) {
+        self.schemas(
+            $.map(response.meta_schemas, function(schema) {
+                return new MetaSchema(schema);
+            })
+        );
+    });
+
+    var getDraftRegistrations = self.getDraftRegistrations();
+
+    getDraftRegistrations.then(function(response) {
+        self.drafts(
+            $.map(response.drafts, function(draft) {
+                return new Draft(draft);
+            })
+        );
+    });
+
+    $.when(getSchemas, getDraftRegistrations).then(function() {
```

Comment:
For one callback (success handler only), consider using `.done` (more readable wrt intent)